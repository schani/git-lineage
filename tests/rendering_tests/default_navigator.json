{
  "active_panel": "Navigator",
  "file_tree": {
    "root": [
      {
        "name": "src",
        "path": "src",
        "is_dir": true,
        "git_status": null,
        "is_expanded": true,
        "children": [
          {
            "name": "lib.rs",
            "path": "src/lib.rs",
            "is_dir": false,
            "git_status": "A",
            "is_expanded": false,
            "children": [],
            "parent_path": "src"
          },
          {
            "name": "main.rs",
            "path": "src/main.rs",
            "is_dir": false,
            "git_status": "M",
            "is_expanded": false,
            "children": [],
            "parent_path": "src"
          }
        ],
        "parent_path": null
      },
      {
        "name": "Cargo.toml",
        "path": "Cargo.toml",
        "is_dir": false,
        "git_status": "M",
        "is_expanded": false,
        "children": [],
        "parent_path": null
      }
    ],
    "repo_root": "",
    "git_status_map": {}
  },
  "selected_file_navigator_index": null,
  "search_query": "",
  "in_search_mode": false,
  "commit_list": [
    {
      "hash": "a1b2c3d4e5f6789012345678901234567890abcd",
      "short_hash": "a1b2c3d",
      "author": "John Doe",
      "date": "2 hours ago",
      "subject": "Add new feature"
    }
  ],
  "selected_commit_index": 0,
  "current_content": [
    "use std::io;",
    "",
    "fn main() {",
    "    println!(\"Hello, world!\");",
    "}"
  ],
  "cursor_line": 0,
  "cursor_column": 0,
  "inspector_scroll_vertical": 0,
  "inspector_scroll_horizontal": 0,
  "show_diff_view": false,
  "status_message": "Ready",
  "is_loading": false,
  "selected_commit_hash": "a1b2c3d4e5f6789012345678901234567890abcd"
}