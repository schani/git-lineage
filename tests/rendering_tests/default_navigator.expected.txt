┌ File Navigator ──────────┐┌ Code Inspector ──────────────────────────────────┐
│▼ src                     ││   1 use std::io;                                 │
│  A lib.rs                ││   2                                              │
│  M main.rs               ││   3 fn main() {                                  │
│M Cargo.toml              ││   4     println!("Hello, world!");               │
│                          ││   5 }                                            │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
└──────────────────────────┘│                                                  │
┌ Commit History ──────────┐│                                                  │
│>> a1b2c3d 2 hours ago Joh││                                                  │
│   More commits available ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
Ready | Tab: Switch panel | 1/2/3: Direct panel focus | []: Older/Younger commit
